src/getBuildingId.py

```diff
@@ -0,0 +1,51 @@
+import json
+import boto3
+import decimal
+from boto3.dynamodb.conditions import Key, Attr
+
+"""Dynamo resource"""
+dynamodb = boto3.resource('dynamodb', region_name = 'us-east-2')
+buildingTable = dynamodb.Table('Buildings')
+
+"""Workaround for formatting issue
+   Source: http://docs.aws.amazon.com/amazondynamodb/latest/developerguide/GettingStarted.Python.04.html
+"""
+class DecimalEncoder(json.JSONEncoder):
+    def default(self, o):
+        if isinstance(o, decimal.Decimal):
+            if o % 1 > 0:
+                return float(o)
+            else:
+                return int(o)
+        return super(DecimalEncoder, self).default(o)
+
+"""Lambda handler function for /buildings/{buildingId} API call
+   Returns building with the buildingId specified in the path
+   or 'Building not found' error if buildingId not found by query search
+"""
+def getBuildingId(event, context):
+    """If buildingId specified, assign it to variable,
+	   use query to find it in building table
+       put it in JSON format and return
+    """
+    buildingId = None
+    if event.get('pathParameters'):
+        buildingId = event.get('pathParameters').get('buildingId')
+		
+        response = buildingTable.query(
```

Comment:
So I know I told you to use `query()`, but after doing a little more digging I realized that `get_item` would actually be more appropriate here, since it returns a single item instead of looking for more than 1. You can switch to `get_item` by replacing this query with `response = buildingTable.get_item(Key={'buildingId': buildingId})`. Then instead of checking the length of `response['Items']` you would say `if response.get('Item')`.

I couldn't tell you whether or not there's a performance difference, if there is one it's probably not significant. This just makes the code a little clearer.